Enable URL and binary PDF for Mistral

pintaf opened the pull request:
I tried to use DocumentURLChunk as it was done for Image, but contrarily to Images where ImageUrl is exported by Mistral sdk, there is no export of DocumentUrl...

This has been properly tested with both kinds of PDFs (binary and URL)

Happy to change to [Mistral]DocumentURLChunk if you tell me how to make it work because I did not succeed...

DouweM commented:
@pintaf Thank you! Looks like the `mistralai` package needs to have its version bumped to get `DocumentUrlChunk`: https://github.com/mistralai/client-python/blob/main/src/mistralai/models/documenturlchunk.py. It would also be good to have a new test for this. You can look at the tests for other models that document support for inspiration.

pintaf commented:
Hi.
I have squashed many useless commits to make it more readable.
I have updated the mistral deps by changing `pyproject.toml`, and running `uv lock`, but this is the first time I use uv, so let me know if any issues.

I added tests, but this is the first time I do tests in pythin, so complete n00b.
First, I had to deal with a `Type "Literal[1704067200] | None" is not assignable to type "int"` error. Most probably due to the mistral package update. Earlier the `MistralChatCompletionResponse` might have allowed None values for `created`, but this is not the case anymore it seems.

In order to add the two tests for binary and URL pdf, I simply compied the tests for binary and URL images from mistral.
I checked how tests were done for example for the gemini model, and I seen they differed significantly.
as @Kludex it seems you participed in both the writing of mistral and gemini model tests, maybe your view on how I designed the test might be valuable.

for the binary PDF, I tried to find a really small binary PDF, and found something in here: https://stackoverflow.com/questions/17279712/what-is-the-smallest-possible-valid-pdf

Looking forward to your remarks.

DouweM commented on tests/models/test_mistral.py:
```diff
@@ -187,13 +188,20 @@ def test_init():
         completion_message(
             MistralAssistantMessage(content='world'),
             usage=MistralUsageInfo(prompt_tokens=1, completion_tokens=1, total_tokens=1),
-            with_created=False,
```
The `coverage` check is complaining that lines 328 and 363 in `mistral.py` are no longer covered by tests, because we're no longer passing `created=None` here. Can you restore this? If you get a type error, we can hide it with a `# pyright: ignore` comment at the end of the line.

DouweM commented on pydantic_ai_slim/pydantic_ai/models/mistral.py:
```diff
@@ -539,10 +540,19 @@ def _map_user_prompt(self, part: UserPromptPart) -> MistralUserMessage:
+                    if item.media_type == 'application/pdf':
+                        content.append(MistralDocumentURLChunk(document_url=item.url, type='document_url'))
+                    else:
+                        raise RuntimeError('DocumentUrl other than PDF is not supported in Mistral.')
```
The `coverage` check is complaining that this line isn't covered in tests. To do so, we can add a test like the existing `test_audio_as_binary_content_input` which hits this error for a non-PDF document (e.g. `text/markdown`)
